### PR TITLE
Add documentation for dotnet-grpc

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Fixes #https://github.com/grpc/grpc-dotnet/issues/437

This covers the dotnet-grpc tooling. Working on the VS tooling piece.

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@JunTaoLuo mention me when you're ready for an edit.

File: aspnetcore/grpc/dotnet-grpc.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+By [John Luo](https://github.com/juntaoluo)
+
+`dotnet-grpc` is a .NET Core global tool for managing Protobuf references within a .NET gRPC project. The tool can be used to add, refresh, remove and list Protobuf references.
+
+## Installation
+
+To install `dotnet-grpc` [.NET Core Global Tool](/dotnet/core/tools/global-tools), run the following command:
+
+```console
```
Comment (scottaddie) [inline aspnetcore/grpc/dotnet-grpc.md]:
```suggestion
```dotnetcli
```

File: aspnetcore/grpc/dotnet-grpc.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+monikerRange: '>= aspnetcore-3.0'
+ms.author: johluo
+ms.date: 09/24/2019
+uid: grpc/dotnet-grpc
+---
+# Manage Protobuf references with dotnet-grpc
+
+By [John Luo](https://github.com/juntaoluo)
+
+`dotnet-grpc` is a .NET Core global tool for managing Protobuf references within a .NET gRPC project. The tool can be used to add, refresh, remove and list Protobuf references.
```
Comment (scottaddie) [inline aspnetcore/grpc/dotnet-grpc.md]:
```suggestion
`dotnet-grpc` is a .NET Core Global Tool for managing Protobuf references within a .NET gRPC project. The tool can be used to add, refresh, remove, and list Protobuf references.
```

File: aspnetcore/grpc/dotnet-grpc.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+## Installation
+
+To install `dotnet-grpc` [.NET Core Global Tool](/dotnet/core/tools/global-tools), run the following command:
+
+```console
+dotnet tool install -g dotnet-grpc
+```
+
+## Add References
```
Comment (scottaddie) [inline aspnetcore/grpc/dotnet-grpc.md]:
```suggestion
## Add references
```

File: aspnetcore/grpc/dotnet-grpc.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+The global tool can be used to add Protobuf references as `<Protobuf />` items to the *.csproj* file:
+
+```xml
+<Protobuf Include="..\Proto\count.proto" GrpcServices="Server" Link="Protos\count.proto" />
+```
+
+The Protobuf references are used to generate the C# client and/or server assets. The tool can be used to create a Protobuf reference from local file(s) on disk or a remote file specified by an URL. The tool will also ensure the correct gRPC package dependencies are added to the project. For example, the `Grpc.AspNetCore` package, which contains gRPC server and client libraries as well as tooling support, will be added to a Web app. Alternatively, the `Grpc.Net.Client`, `Grpc.Tools` and `Google.Protobuf` packages, which contains only the gRPC client libraries and tooling support, will be added to a Console app.
+
+### Add File
```
Comment (scottaddie) [inline aspnetcore/grpc/dotnet-grpc.md]:
```suggestion
### Add file
```

File: aspnetcore/grpc/dotnet-grpc.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+The `add-file` command is used to add local file(s) on disk as Protobuf reference(s). The file path(s) provided can be relative to the current directory or absolute paths and may contain wild cards for pattern based file globbing. If any file(s) are outside the project directory, a `Link` element will be added to display the file under the folder `Protos` in Visual Studio.
+
+### Usage
+
+```console
```
Comment (scottaddie) [inline aspnetcore/grpc/dotnet-grpc.md]:
```suggestion
```dotnetcli
```

File: aspnetcore/grpc/dotnet-grpc.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+| -i | --additional-import-dirs | Additional directories to be used when resolving imports for the protobuf files. This is a semicolon separated list of paths.
+| | --access | The access modifier to use for the generated C# classes. Default value is Public. Accepted values are Internal and Public.
+
+## Remove
+
+The `remove` command is used to remove Protobuf references from the *.csproj* file. The command accepts path(s) arguments and source URL as arguments. Note that the tool only removes the Protobuf reference but the actual *.proto* file is not deleted, even if it were originally downloaded from a remote URL.
+
+### Usage
+
+```console
```
Comment (scottaddie) [inline aspnetcore/grpc/dotnet-grpc.md]:
```suggestion
```dotnetcli
```

File: aspnetcore/grpc/dotnet-grpc.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+|-|-|-|
+| -p | --project | The path to the project file to operate on. If a file is not specified, the command will search the current directory for one.
+
+## Refresh
+
+The `refresh` command is used to update a remote reference with the latest content from the source URL. Both the download file path and the source URL can be used to specify the reference to be updated. Note that the hashes of the file contents is compared to determine whether the local file should be updated and no timestamp information is compared. The tool will always replace the the local file with the remote file if an update is needed.
+
+### Usage
+
+```console
```
Comment (scottaddie) [inline aspnetcore/grpc/dotnet-grpc.md]:
```suggestion
```dotnetcli
```

File: aspnetcore/grpc/dotnet-grpc.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+| -p | --project | The path to the project file to operate on. If a file is not specified, the command will search the current directory for one.
+| | --dry-run | Output a list of file(s) that will be updated without downloading any new content.
+
+## Refresh
+
+The `list` command is used to diplay all the Protobuf references in the project file. Note that if all values of a column are default values, the column may not be omitted.
+
+### Usage
+
+```console
```
Comment (scottaddie) [inline aspnetcore/grpc/dotnet-grpc.md]:
```suggestion
```dotnetcli
```

File: aspnetcore/grpc/dotnet-grpc.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+---
+# Manage Protobuf references with dotnet-grpc
+
+By [John Luo](https://github.com/juntaoluo)
+
+`dotnet-grpc` is a .NET Core global tool for managing Protobuf references within a .NET gRPC project. The tool can be used to add, refresh, remove and list Protobuf references.
+
+## Installation
+
+To install `dotnet-grpc` [.NET Core Global Tool](/dotnet/core/tools/global-tools), run the following command:
```
Comment (scottaddie) [inline aspnetcore/grpc/dotnet-grpc.md]:
```suggestion
To install the `dotnet-grpc` [.NET Core Global Tool](/dotnet/core/tools/global-tools), run the following command:
```

**Comment by JunTaoLuo** [comment]
cc @Rick-Anderson

**Comment by Rick-Anderson** [comment]
EVERYBODY HOLD OFF EDITs while I PR this at #14935

**Comment by JunTaoLuo** [comment]
Looks like this is updated. Anything else before we merge this?

**Comment by Rick-Anderson** [comment]
Sorry, I forgot to S&M after merging my branch.

**Pull request merged**